source/extensions/common/tap/tap_config_base.cc

```diff
@@ -277,6 +281,90 @@ void FilePerTapSink::FilePerTapSinkHandle::submitTrace(
   }
 }
 
+// The implemented code for UDP sink
+UdpTapSink::UdpTapSink(const envoy::extensions::tap_sinks::udp_sink::v3::UdpSink& config)
+    : config_(config) {
+  if (config_.udp_address().protocol() != envoy::config::core::v3::SocketAddress::UDP) {
+    ENVOY_LOG_MISC(warn, "{}: Only suport UDP and invalid protocol", __func__);
+    return;
+  }
+
+  // Verify the address (ipv4/ipv6)
+  udp_server_address_ = Network::Utility::parseInternetAddressNoThrow(
+      config_.udp_address().address(), static_cast<uint16_t>(config_.udp_address().port_value()),
+      false);
+  if (!udp_server_address_) {
+    ENVOY_LOG_MISC(warn, "{}: Invalid configuration for address {} or port_value {}", __func__,
+                   config_.udp_address().address().c_str(), config_.udp_address().port_value());
+    return;
+  }
+
+  // Create socket
+  udp_socket_ =
+      std::make_unique<Network::SocketImpl>(Network::Socket::Type::Datagram, udp_server_address_,
+                                            nullptr, Network::SocketCreationOptions{});
+
+  // Create udp writer
+  udp_packet_writer_ = std::make_unique<Network::UdpDefaultWriter>(udp_socket_->ioHandle());
+  ENVOY_LOG_MISC(debug, "{}: UDP packet writer is created", __func__);
+}
+
+UdpTapSink::~UdpTapSink() { ENVOY_LOG_MISC(warn, "{}: UDP UdpTapSink() is called", __func__); }
+
+void UdpTapSink::UdpTapSinkHandle::submitTrace(TraceWrapperPtr&& trace,
+                                               envoy::config::tap::v3::OutputSink::Format format) {
+  switch (format) {
+    PANIC_ON_PROTO_ENUM_SENTINEL_VALUES;
+  case envoy::config::tap::v3::OutputSink::PROTO_BINARY:
+  case envoy::config::tap::v3::OutputSink::PROTO_BINARY_LENGTH_DELIMITED:
```

Comment:
If the fall through is explicit here, please add `FALLTHRU;` or `ABSL_FALLTHROUGH_INTENDED;` (see style guide).

Reply:
Nice comment, changed